includes/emails/class-email.php

```diff
@@ -83,6 +83,32 @@ public static function init( $meta = [] ) {
 			$meta = apply_filters( 'hivepress/v1/emails/' . hp\get_class_name( $class ) . '/meta', $meta );
 		}
 
+		foreach ( hp\get_array_value( $meta, 'tokens', [] ) as $token ) {
```

Comment:
Please move this logic to the Email component (using a callback for the above hook), we keep the base classes as standard as possible, and custom logic is moved to components (e.g. we do the same with Forms, reCaptcha is integrated from the component using hooks instead of the base class).